src/daemon/protocols/edp.c

```diff
@@ -353,6 +356,7 @@ edp_decode(struct lldpd *cfg, char *frame, int s, struct lldpd_hardware *hardwar
 				goto malformed;
 			}
 			port->p_id_len = strlen(port->p_id);
+			free(port->p_descr);
```

Comment:
Fix a memory leak.
A memory leak occurs when `case EDP_TLV_INFO` is called twice.

Scenario:
1- Call occurs.
-----> Memory leak.
2- Call occurs.
-----> Memory is freed by the program.

Therefore, if there are two or more calls to `case EDP_TLV_INFO`, all of them leak memory, except the last one.
This change handles all memory leaks.